### PR TITLE
feat(KPop) Adds positions & theme

### PR DESCRIPTION
**Pull request by travispamaral**
### Summary
This PR adds all popper.js supported positions (-start, -end) to supported positions. It also updates the class prop to not remove `.k-popover` when passed. This also adds css padding theming. Additionally I added a [setting](https://github.com/Kong/kongponents/compare/feat/kpop-add-more-positions?expand=1#diff-198bd74514b5965edfd8dd6f5317205fR256) to ensure the pop over does not flow outside of the viewport ever.

#### Changes made:
* Updated placement prop
* Updated class logic
* Adds `KPopPaddingX` & `KPopPaddingY`

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

**Comment by darrenjennings** [comment]
Can you add some docs? maybe a dropdown of all the available positions and a docs user can switch them

**Comment by travispamaral** [comment]
> Can you add some docs? maybe a dropdown of all the available positions and a docs user can switch them

I don’t follow. Like a dynamic preview of positions?

**Comment by darrenjennings** [comment]
yea so you can pick which position you want in the preview or do like this: 
https://bootstrap-vue.js.org/docs/components/popover under "<b-popover> Component basic usage"

File: docs/components/popover.md [hunk 1]
```diff
@@ -300,6 +346,22 @@ triggers)
   export default {
     data () {
       return {
+        selectedPosition: 'auto',
```
Comment (travispamaral) [inline docs/components/popover.md]: these are added to this block because vuepress isn't picking up the data if its added somewhere else in the doc

**Pull request merged**